### PR TITLE
Chess: A chess game for serenity

### PR DESCRIPTION
**Pull request by Petelliott**
I'm writing a hopefully fully featured chess app for serenity. In this PR i'm just adding the UI and controls, but I will add engine support and a serenity-native engine in the future. If anyone would like to make a better piece set you are more than welcome.

- [x] Dragging controls 
- [x] Moves
  - [x] Normal moves
  - [x] Castling
  - [x] Pawn promotion
  - [x] En passant
- [x] rules
  - [x] Checkmate
  - [x] Stalemate
  - [x] Threefold/Fivefold repitition
  - [x] Fifty move rule
  - [x] Insufficient material
- [x] allow user to select piece sets and board colours

2020-08-11-194817_3200x1800_scrot

### CODE REVIEW

**Comment by stelar7** [comment]
Made some better icons for you :)
You can download them here; https://www.dropbox.com/sh/6xen9yj4ut0c5ls/AAD4mK5fn0vvwE8A7BmbtHWma

File: Games/Chess/Chess.cpp [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        int rank_delta = move.to.rank - move.from.rank;
+        int file_delta = move.to.file - move.from.file;
+        if (abs(rank_delta) <= 1 && abs(file_delta) <= 1) {
+            if (get_piece(move.to).colour != colour) {
+                return true;
+            }
+        }
+
+        if (colour == Colour::White) {
+            if ((move.to == Square("a1") || move.to == Square("c1")) && m_white_can_castle_queenside && get_piece(Square("b1")).type == Type::None && get_piece(Square("c1")).type == Type::None && get_piece(Square("d1")).type == Type::None) {
```
Comment (stelar7) [inline Games/Chess/Chess.cpp:225]: You need to check that the squares the king moves over are not targeted by an enemy piece too :)

Comment (Petelliott) [reply Games/Chess/Chess.cpp:249]: nice catch :^). this is now implemented in `is_legal` rather than `is_legal_no_check` to prevent infinite recursion.

**Comment by awesomekling** [comment]
Very rad :)

**Pull request merged**